qiskit_ibm/accountprovider.py

```diff
@@ -139,6 +140,7 @@ def _backends(self) -> Dict[str, IBMQBackend]:
         """
         if not self.__backends:
             self.__backends = self._discover_remote_backends()
+            self._backend.discover_backends()
```

Comment:
`discover_backends()` adds each backend as an attribute to `IBMQBackendService`, but `_discover_remote_backends` also does that (line 224)? This will cause `discover_backends()` to append `_` to the name and add a duplicate.